jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -28,24 +30,72 @@ class Catalogs(PluginTemplateMixin, ViewerSelectMixin):
     template_file = __file__, "catalogs.vue"
     catalog_items = List([]).tag(sync=True)
     catalog_selected = Unicode("").tag(sync=True)
+    from_file = Unicode().tag(sync=True)
+    from_file_message = Unicode().tag(sync=True)
     results_available = Bool(False).tag(sync=True)
     number_of_results = Int(0).tag(sync=True)
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        # TODO: Add more catalogs.
+
         self.catalog = SelectPluginComponent(self,
                                              items='catalog_items',
                                              selected='catalog_selected',
-                                             manual_options=['SDSS'])
+                                             manual_options=['SDSS', 'From File...'])
+
+        # file chooser for From File
+        start_path = os.environ.get('JDAVIZ_START_DIR', os.path.curdir)
+        self._file_upload = FileChooser(start_path)
+        self.components = {'g-file-import': self._file_upload}
+        self._file_upload.observe(self._on_file_path_changed, names='file_path')
+        self._cached_table_from_file = {}
+
+    def _on_file_path_changed(self, event):
+        self.from_file_message = 'Checking if file is valid'
+        path = event['new']
+        if (path is not None
+                and not os.path.exists(path)
+                or not os.path.isfile(path)):
+            self.from_file_message = 'File path does not exist'
+            return
+
+        try:
+            table = QTable.read(path)
+        except Exception:
+            self.from_file_message = 'Could not parse file with astropy.table.QTable.read'
+            return
+
+        if 'sky_centroid' not in table.keys():
```

Comment:
I wonder if the column name can be changed by the user but maybe that can be future work.